From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harmer <mark.harmer@discordapp.com>
Date: Mon, 27 Feb 2023 12:56:15 -0800
Subject: Enabling AudioVideoTracks feature to use HTMLMediaElements
 audioTracks.

enables audiotrack feature

diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index 75e44e6b7cdd95c1e39b3759939ec1146232f255..98c5e4c1209f95a05ff474c44ca5eb7d18a0d067 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -405,7 +405,7 @@
     },
     {
       name: "AudioVideoTracks",
-      status: "experimental",
+      status: "stable",
     },
     {
       name: "AutoDarkMode",
