From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harmer <mark.harmer@discordapp.com>
Date: Mon, 27 Feb 2023 12:56:15 -0800
Subject: Enabling AudioVideoTracks feature to use HTMLMediaElements
 audioTracks.

enables audiotrack feature

diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index 5f91aa499ae04a5a21135b7345747cdad694bd21..ecd0587aa4b94b59dfb6011ea6bd2a24bfecff66 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -357,7 +357,7 @@
     },
     {
       name: "AudioVideoTracks",
-      status: "experimental",
+      status: "stable",
       base_feature: "none",
     },
     {
