From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harmer <mark.harmer@discordapp.com>
Date: Wed, 4 Oct 2023 22:01:26 -0700
Subject: Use 32-bit PartitionAlloc sizes for 64-bit builds


diff --git a/base/allocator/partition_allocator/partition_alloc_config.h b/base/allocator/partition_allocator/partition_alloc_config.h
index 7d3934b8f4aef668248a92d33e9178a68e64fae5..68d724f31d031b3864c99d4ba42299ceab110d1a 100644
--- a/base/allocator/partition_allocator/partition_alloc_config.h
+++ b/base/allocator/partition_allocator/partition_alloc_config.h
@@ -30,7 +30,7 @@ static_assert(sizeof(void*) != 8, "");
 #define PA_STARSCAN_NEON_SUPPORTED
 #endif
 
-#if defined(PA_HAS_64_BITS_POINTERS) && (BUILDFLAG(IS_IOS) || BUILDFLAG(IS_WIN))
+#if defined(PA_HAS_64_BITS_POINTERS) && BUILDFLAG(IS_IOS)
 // Allow PA to select an alternate pool size at run-time before initialization,
 // rather than using a single constexpr value.
 //
diff --git a/base/allocator/partition_allocator/partition_alloc_constants.h b/base/allocator/partition_allocator/partition_alloc_constants.h
index 931cf2941cad5f0027dc74640c8bb532e4c84ea2..88349813df28545baa6fd5cb06c1eb527cd1e06d 100644
--- a/base/allocator/partition_allocator/partition_alloc_constants.h
+++ b/base/allocator/partition_allocator/partition_alloc_constants.h
@@ -272,6 +272,9 @@ constexpr size_t kNumPools = 3;
 // unrealistic as of 2022.
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
 constexpr size_t kPoolMaxSize = 8 * kGiB;
+#elif BUILDFLAG(IS_WIN)
+// Limit size for Windows 64-bit builds to 4GB
+constexpr size_t kPoolMaxSize = 4 * kGiB;
 #else
 constexpr size_t kPoolMaxSize = 16 * kGiB;
 #endif
